src/scroll.js

```diff
@@ -0,0 +1,41 @@
+const AlpineScrollMagicMethod = {
+    start() {
+
+        Alpine.addMagicProperty('scroll', function ($el) {
+            return function (target) {
+                var isSmoothScrollSupported = 'scrollBehavior' in document.documentElement.style;
+
+                if (Number.isInteger(target)) {
+                    isSmoothScrollSupported
+                        ? window.scroll({
+                            top: target,
+                            behavior: 'smooth'
+                        })
+                        : window.scroll(0, target);
+                    return
+                }
+
+                if (typeof target === 'string') {
+                    target = document.querySelector(target)
```

Comment:
I'm not sure. The standard syntax for reference in alpine is $refs.foo (or $ref['foo']). I don't want to create confusion because it wouldn't be immediate that we are talking about Alpine references and if people get used to it, they might try to apply the same concept in different places (alpine itself?) raising unnecessary bugs on other projects.
To be fair, I'm also torn about supporting query selectors as strings, maybe we should support $parent and $component instead.

Since $scroll already supports element, the query selector can be then achieved with `$scroll(document.querySelctor('#foo'))` or maybe in the future with another helper like `$scroll($selectEl('#foo'))`.

On the same topic, the integer is there again only for 'scroll to top', which is the only real use case I can talk about, another options would be to drop it and maybe only support $scrollToTop(). (A bit unsure on the last one, someone will need it at some point).